NPE in InstanceService.getAvailableJobInstances

## Bug Report

throw NPE in [InstanceService.getAvailableJobInstances()](https://github.com/apache/shardingsphere-elasticjob/blob/master/elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/instance/InstanceService.java#L73)  when upgrade from 3.0.0-RC1 to 3.0.0.
```
2021-08-02 11:56:20.014 ERROR [AlertSendJob_Worker-1] [{}] Job (DEFAULT.AlertSendJob threw an exception. (QuartzScheduler.java:2407)org.quartz.core.ErrorLogger.schedulerError
org.quartz.SchedulerException: Job threw an unhandled exception.
	at org.quartz.core.JobRunShell.run(JobRunShell.java:213) ~[quartz-2.3.2.jar:?]
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573) ~[quartz-2.3.2.jar:?]
Caused by: java.lang.NullPointerException
	at org.apache.shardingsphere.elasticjob.lite.internal.instance.InstanceService.getAvailableJobInstances(InstanceService.java:73) ~[elasticjob-lite-core-3.0.0.jar:3.0.0]
	at org.apache.shardingsphere.elasticjob.lite.internal.sharding.ShardingService.shardingIfNecessary(ShardingService.java:111) ~[elasticjob-lite-core-3.0.0.jar:3.0.0]
	at org.apache.shardingsphere.elasticjob.lite.internal.schedule.LiteJobFacade.getShardingContexts(LiteJobFacade.java:113) ~[elasticjob-lite-core-3.0.0.jar:3.0.0]
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.execute(ElasticJobExecutor.java:85) ~[elasticjob-executor-kernel-3.0.0.jar:3.0.0]
	at org.apache.shardingsphere.elasticjob.lite.internal.schedule.LiteJob.execute(LiteJob.java:35) ~[elasticjob-lite-core-3.0.0.jar:3.0.0]
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202) ~[quartz-2.3.2.jar:?]
	... 1 more
```

```
elasticjob.jobs.AlertSendJob.elastic-job-class=xxxcenter.job.SendJob
elasticjob.jobs.AlertSendJob.cron=0/5 * * * * ?
elasticjob.jobs.AlertSendJob.sharding-total-count=2
```

--- a/elasticjob-infra/elasticjob-registry-center/src/main/java/org/apache/shardingsphere/elasticjob/reg/zookeeper/ZookeeperRegistryCenter.java
+++ b/elasticjob-infra/elasticjob-registry-center/src/main/java/org/apache/shardingsphere/elasticjob/reg/zookeeper/ZookeeperRegistryCenter.java
@@ -137,10 +137,7 @@ public final class ZookeeperRegistryCenter implements CoordinatorRegistryCenter 
             return getDirectly(key);
         }
         Optional<ChildData> resultInCache = cache.get(key);
-        if (resultInCache.isPresent()) {
-            return null == resultInCache.get().getData() ? null : new String(resultInCache.get().getData(), StandardCharsets.UTF_8);
-        }
-        return getDirectly(key);
+        return resultInCache.map(v -> null == v.getData() ? null : new String(v.getData(), StandardCharsets.UTF_8)).orElse(getDirectly(key));
     }
     
     private CuratorCache findCuratorCache(final String key) {

--- a/elasticjob-infra/elasticjob-registry-center/src/main/java/org/apache/shardingsphere/elasticjob/reg/zookeeper/ZookeeperRegistryCenter.java
+++ b/elasticjob-infra/elasticjob-registry-center/src/main/java/org/apache/shardingsphere/elasticjob/reg/zookeeper/ZookeeperRegistryCenter.java
@@ -137,7 +137,7 @@ public final class ZookeeperRegistryCenter implements CoordinatorRegistryCenter 
             return getDirectly(key);
         }
         Optional<ChildData> resultInCache = cache.get(key);
-        return resultInCache.map(v -> null == v.getData() ? null : new String(v.getData(), StandardCharsets.UTF_8)).orElse(getDirectly(key));
+        return resultInCache.map(v -> null == v.getData() ? null : new String(v.getData(), StandardCharsets.UTF_8)).orElseGet(() -> getDirectly(key));
     }
     
     private CuratorCache findCuratorCache(final String key) {

--- a/elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/instance/InstanceService.java
+++ b/elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/instance/InstanceService.java
@@ -70,7 +70,7 @@ public final class InstanceService {
         List<JobInstance> result = new LinkedList<>();
         for (String each : jobNodeStorage.getJobNodeChildrenKeys(InstanceNode.ROOT)) {
             JobInstance jobInstance = YamlEngine.unmarshal(jobNodeStorage.getJobNodeData(instanceNode.getInstancePath(each)), JobInstance.class);
-            if (serverService.isEnableServer(jobInstance.getServerIp())) {
+            if (null != jobInstance && serverService.isEnableServer(jobInstance.getServerIp())) {
                 result.add(new JobInstance(each));
             }
         }

